### ISSUE TITLE
Question - is it possible to decompile code of FabricCommon.dll ?

### ISSUE THREAD

**Issue by nareshkhatri81** [Azure/service-fabric-issues#993]
Hi Team,

i need to see how encyption/decryption is done inside FabirCommon.dll.
But i am not able to decompile it.
i have one issue where i am not able to decrypt secrets inside containers.
i need to understand algorithm used by fabric and do it with C# inside containers.
is source code is available somewhere in github ?
i am not sure if service fabric code is fully available on github.

https://github.com/Azure/service-fabric-issues/issues/975

Thanks,
Naresh Khatri

**Comment by nareshkhatri81** [comment]
[DllImport("FabricCommon.dll", PreserveSig = false)]
		[return: MarshalAs(UnmanagedType.Interface)]
		internal static extern NativeCommon.IFabricStringResult FabricDecryptText([In] IntPtr encryptedValue, [In] NativeTypes.FABRIC_X509_STORE_LOCATION certStoreLocation);

it seems it calls some native code. 
can someone help which native api its calling ?

**Comment by raunakpandya** [comment]
We also have C# API which you can use to encrypt and decrypt in your C# code - https://msdn.microsoft.com/en-us/library/system.fabric.security.encryptionutility.encrypttext.aspx

Regarding code for FabricCommon, its open source now - :) Look here - https://github.com/Microsoft/service-fabric/blob/b28fe85931de297ac3791a2c8099e62fcde3b60e/src/prod/src/retail/native/FabricCommon/dll/dllmain.cpp#L321

**Comment by raunakpandya** [comment]
Closing issue

**Comment by nareshkhatri81** [comment]
@raunakpandya  - I created sample Console application in C#.

Added Below Package
Install-Package Microsoft.ServiceFabric

 string EncryptedText = EncryptionUtility.EncryptText("Test", "‎ef5d044856ccb15a92f819864597d1cbe34b727b", @"CurrentUser\Personal\Certificates");

                StoreLocation storelocation = StoreLocation.CurrentUser;
                System.Security.SecureString DecryptText = EncryptionUtility.DecryptText(EncryptedText, storelocation);

Getting below error.

An unhandled exception of type 'System.BadImageFormatException' occurred in Unknown Module.
Could not load file or assembly 'System.Fabric, Version=6.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. An attempt was made to load a program with an incorrect format.

Thanks,
Naresh Khatri
